backend/kernelCI_app/models.py

```diff
@@ -228,3 +234,102 @@ class Meta:
         ]
+
+
+class HardwareStatus(models.Model):
+    pk = models.CompositePrimaryKey(
+        "test_origin",
+        "platform",
+        "checkout_id",
+    )
+    checkout_id = models.TextField()
+    test_origin = models.CharField(max_length=100)
+    platform = models.CharField(max_length=100)
+
+    compatibles = ArrayField(models.TextField(), null=True)
+    start_time = models.DateTimeField()
+
+    # build status
+    build_pass = models.IntegerField(default=0)
+    build_failed = models.IntegerField(default=0)
+    build_inc = models.IntegerField(default=0)
+
+    # boot status
+    boot_pass = models.IntegerField(default=0)
+    boot_failed = models.IntegerField(default=0)
+    boot_inc = models.IntegerField(default=0)
+
+    # test status
+    test_pass = models.IntegerField(default=0)
+    test_failed = models.IntegerField(default=0)
+    test_inc = models.IntegerField(default=0)
+
+    class Meta:
+        db_table = "hardware_status"
+        indexes = [
+            models.Index(
+                fields=["test_origin", "start_time"], name="hw_status_origin_time"
+            ),
+            models.Index(fields=["checkout_id"], name="hw_status_checkout_id"),
+        ]
+
+
+class LatestCheckout(models.Model):
+    checkout_id = models.TextField(primary_key=True)
+    start_time = models.DateTimeField()
+
+    origin = models.CharField(max_length=100)
+    tree_name = models.TextField(null=True, blank=True)
+    git_repository_url = models.TextField(null=True, blank=True)
+    git_repository_branch = models.TextField(null=True, blank=True)
+
+    class Meta:
+        db_table = "latest_checkout"
+        constraints = [
+            models.UniqueConstraint(
+                fields=[
+                    "origin",
+                    "tree_name",
+                    "git_repository_url",
+                    "git_repository_branch",
+                ],
+                name="latest_checkout_unique",
+                nulls_distinct=False,
+            ),
+        ]
+
+
+class PendingTest(models.Model):
+    test_id = models.TextField(primary_key=True)
+    origin = models.CharField(max_length=100)
+    platform = models.CharField(max_length=100)
+    compatible = ArrayField(models.TextField(), null=True)
+    build_id = models.TextField()
+    status = models.CharField(max_length=1, choices=SimplifiedStatusChoices.choices)
+    is_boot = models.BooleanField()
+
+    class Meta:
+        db_table = "pending_test"
+
+
+class HardwareStatusEntityType(models.TextChoices):
+    TEST = "T"
+    BUILD = "B"
+
+
+class ProcessedHardwareStatus(models.Model):
+    pk = models.CompositePrimaryKey(
+        "hardware_key",
+        "entity_id",
+        "entity_type",
+    )
+    hardware_key = models.BinaryField(
+        max_length=32
+    )  # this holds a sha256, thus digest_size = 32 bytes
+    entity_id = models.TextField()
+    entity_type = models.CharField(
+        max_length=1, choices=HardwareStatusEntityType.choices
+    )
```

Comment:
we should include entity_id/type inside the `hardware_key`, currently the lookup for `hardware_key` will return all entities ever processed for that hardware, as the system grows and we never delete `ProcessedHardwareStatus` entries, the set will be large and this will hurt performance (and memory usage).

maybe we should keep a `checkout_id` field from a cron job we should remove those that are not related to the latest checkouts, same as we did with `delete_unused_hardware_...` (or even do it from inside that job)